File: pset6/caesar/caesar.py
# Caesar's algorithm

import sys
from cs50 import get_string

# Check if there exist two arguments
if len(sys.argv) != 2:
    print("You have to pass TWO arguments!")
    sys.exit(1)

# Get the key
k = int(sys.argv[1])
# Prompt user to enter plain text
plainText = get_string("plaintest: ")
chiperText = ""

# Iterate over each character in plain text
for i in range(len(plainText)):
    char = plainText[i]
    if char.isalpha():
        if char.islower():
            # ord("a") equals 97
            char = chr((((ord(plainText[i]) - 97) + k) % 26) + 97)
        else:
            # ord("A") equals 65
            char = chr((((ord(plainText[i]) - 65) + k) % 26) + 65)
    chiperText += char

print(f"ciphertext: {chiperText}")

File: pset2/caesar/caesar.c
//caesar's algorithm
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("You have to pass TWO arguments!\n");
        return 1;
    }

    //convert key to int
    int k = atoi(argv[1]);
    string plainText = get_string("plaintext: ");

    for (int i = 0, n = strlen(plainText); i < n; i++)
    {
        //check if i'th character is alpha
        if (isalpha(plainText[i]))
        {
            if (isupper(plainText[i]))
            {
                //convert character to it's alphabetical index
                //and at the finish convert in to it's ASCII code
                plainText[i] = (((plainText[i] - 'A') + k) % 26) + 'A';
            }
            else if (islower(plainText[i]))
            {
                plainText[i] = (((plainText[i] - 'a') + k) % 26) + 'a';
            }
        }
    }
    //plainText now converted to chiperText
    printf("ciphertext: %s\n", plainText);
}
